### PR TITLE
Add BikeAnnotation clustering

### PR DESCRIPTION
**Pull request by mltbnz**
## 📝 Docs

- [x] Did you update the [CHANGELOG.md](https://github.com/criticalmaps/criticalmaps-ios/blob/master/CHANGELOG.md)?

## 📲 What

Add `BikeClusterAnnotationView` and some minor fixes.

## 🤔 Why

Clustering a map is a good way of visually structuring and it makes the map easier to `read`.
The default zoomLevel is a good fit IMO. Also I consider this a low hanging fruit

## 👀 See

| Before 🐛 | After 🦋 |
| --- | --- |
| Screenshot 2020-03-20 at 15 19 52 | Screenshot 2020-03-20 at 14 56 06 |

### CODE REVIEW

File: CriticalMass/BikeAnnotationView.swift [hunk 1]
```diff
@@ -41,82 +41,54 @@ class BikeAnnoationView: MKAnnotationView {
...
-        didSet {
-            friendView?.name = friend?.name ?? ""
+        if #available(iOS 11.0, *) {
+            clusteringIdentifier = BikeClusterAnnotationView.clusteringIdentifier
         }
     }
+}
+
+@available(iOS 11.0, *)
+class BikeClusterAnnotationView: MKAnnotationView {
```
Comment (fbernutz) [inline CriticalMass/BikeAnnotationView.swift]: the diff is super difficult to read 🙈 You basically added the `BikeClusterAnnotationView` and moved `FriendAnnotationView` and `FriendView` into an own file, right?

**Comment by lennet** [comment]
How does it look like for a mass? Can you still see the snake?

**Comment by mltbnz** [comment]
Sure. It’ll only cluster for far out zoom levels. I’ll add a gif to demonstrate

File: CriticalMass/BikeAnnotationView.swift (hunk 1 above)
Comment (mltbnz) [reply CriticalMass/BikeAnnotationView.swift]: Yeah. That was probably not the best idea

Comment (mltbnz) [reply CriticalMass/BikeAnnotationView.swift]: Yeah. That was probably not the best idea

Comment (mltbnz) [reply CriticalMass/BikeAnnotationView.swift]: moved it the cluster annotation view to its own file 📁

**Comment by mltbnz** [comment]
> How does it look like for a mass? Can you still see the snake if you zoom in?

But it could actually be a problem since the cluster is based on a collision rect. So you’d have some cluster here and there and others would be displayed as single dots. Maybe this wasn’t such a low hanging fruit. Just needed to move my thoughts away from this corona craziness.

**Comment by mltbnz** [comment]
But I think this can be tweeked to only form a cluster when a certain zoom level was reached 🛠🔦

**Comment by mltbnz** [comment]
So I think the mapviews clustering is not the right choice to give users an overview of how many people are attending. The collisionMode cant be tweaked or set via zoomLevel. I'll close this 🤷‍♂️

**Pull request closed without merging**